client/homebrew/editor/snippetbar/snippets/cardtemplate.gen.js

```diff
@@ -0,0 +1,158 @@
+const _ = require('lodash');
+
+module.exports = function(){
+
+	return [
+		'<style>',
```

Comment:
This is a quite a significant snippet to insert and I'm not the biggest fan of putting a lot of CSS into what is supposed to be "only markdown" (but I do understand that this is sometimes impossible to avoid).

My question here: is it possible to apply the same styles not to `.page`, but to another `div`? I mean what if we ask users to create a `<div class="deck-container">...</div>` (like for example it is done with wide blocks) and put all of these styles into built-in homebrewery stylesheets instead of adding them directly to the page source?

Reply:
On reflection, this is effectively a full-blown template, so it may not be appropriate to include it as such. It would likely benefit from being a template that can be applied in the separate CSS tab in the new editor, so that there is a clearer delineation between the CSS and the Markdown - the Markdown actually becomes quite simple once these CSS changes have been applied.

[Example with zoom and page layout](https://homebrewery.naturalcrit.com/share/1EbOq21BlSlvS-8bSvehk4jfzKINUYfhYVhimiKzT1Q2N)  
[Example without zoom/layout](https://homebrewery.naturalcrit.com/share/1XSOHsSodcSnv_UnVDxQOjMHYiVUCNYCuTylhQjnqbpTm)

So at this point, it may be best to put it on hold until the new Renderer is complete and the editor has a separate CSS tab.

Reply:
I thought about this some more and came around to @AlexeySachkov's way of thinking. While the original is great for online viewing or creating an entire deck of magic items, it doesn't print to PDF well and it is not possible to embed a card in a larger document.  
After some manipulation, [this Homebrew](https://homebrewery.naturalcrit.com/share/140NrG9yBcfdlph10Bt2vs_R5Q-OQJY-rKL3TnfK7eqbR) is the result.

Unfortunately the Markdown parser doesn't parse inside `div`s, so I had to replicate the styles in their own classes, which makes it a little more opaque to the standard user, although this can probably be simplified by duplicating the class in the CSS build files.

The best way forward for this version of the card generator is probably to embed the `.card` style in the built CSS, and add a snippet that provides an example of creating a card with the various new div classes. I'll push a commit through for further review in the near future.

Reply:
> Unfortunately the Markdown parser doesn't parse inside divs, so I had to replicate the styles in their own classes, which makes it a little more opaque to the standard user, although this can probably be simplified by duplicating the class in the CSS build files.

This seems a bit strange to me, could you please outline this into a separate issue with some example so we could try to analyze and fix it later?

Reply:
We have a custom rule set up so that markdown does indeed parse inside of divs. You can see it working on the live site such as in the "descriptive text box" snippet:
```
<div class='descriptive'>
##### Time to Drop Knowledge
Use notes to point out some interesting information. 

**Tables and lists** both work within a note.
</div>
```